samples/attestation/host/host.cpp

```diff
@@ -150,6 +191,56 @@ int main(int argc, const char* argv[])
         return 1;
     }
 
+    if (argc == 5)
+    {
+        /* Set logging callback */
+        FILE* out_file = fopen(argv[4], "w");
+        oe_log_set_callback((void*)out_file, host_logging_callback);
+
+        int oe_log_level_size, azdcap_log_level_size;
+
+        /* Set logging levels */
+#if defined(__linux__)
+        /* Get current values of OE_LOG_LEVEL and AZDCAP_DEBUG_LOG_LEVEL */
+        char* oe_log_level = getenv(OE_LOG_LEVEL);
+        char* azdcap_debug_log_level = getenv(AZDCAP_DEBUG_LOG_LEVEL);
+
+        /* Set log level "WARNING" for OE_LOG_LEVEL and AZDCAP_DEVUG_LOG_LEVEL
+         */
+        if (setenv(OE_LOG_LEVEL, "WARNING", 1) != 0)
+        {
+            printf("Failed to set environment variable 'OE_LOG_LEVEL'\n");
+            return 1;
+        }
+        if (setenv(AZDCAP_DEBUG_LOG_LEVEL, "WARNING", 1) != 0)
+        {
+            printf("Failed to set environment variable "
+                   "'AZDCAP_DEBUG_LOG_LEVEL'\n");
+            return 1;
+        }
+#elif defined(_WIN32)
+        /* Get set values of OE_LOG_LEVEL and AZDCAP_DEBUG_LOG_LEVEL */
+        oe_log_level_size =
```

Comment:
If restoring code is not needed, consider making `oe_log_level`,  `oe_log_level_size` &c localize to here.

Reply:
In the sample we are explicitly setting the values of environment variables OE_LOG_LEVEL and AZDCAP_DEBUG_LOG_LEVEL to WARN. The restoration is needed, in case the environment variables are set previously by the user.

Also when OE tests are run, if the values are not restored, the logs would have WARN traces for all tests run after this sample and might break tests.

Reply:
Just to make sure, can you double check the scope of environment variable setting APIs?
From my understanding, the environment variables will be copied when the process starts. Changing the environment within the process should just affect the process itself (i.e., its own copy) instead of the global ones. I'm not sure if Windows has different behavior than Linux.

If the above is not the case, you can similar approach as
https://github.com/openenclave/openenclave/blob/b497b8b121879c222753057e2e490ffc42deb502/tests/sgx/backtrace/CMakeLists.txt#L20
to set the environment for the particular test only.

Reply:
I checked, the APIs only set environment variables for only the current process. I am removing env variable setting logic inside code. Moved to CMakeList.txt.